### PR TITLE
feat: add golint ci

### PR DESCRIPTION
**Pull request by hanshal101**
## Feature
Adds the Golang Lint using the `golangci-lint` in the CI

### CODE REVIEW

**Comment by hanshal101** [comment]
Hey @dr-housemd not sure why did the CI failed, I have used this CI on other orgs as well but never showed this error.
I think this might be a `timeout` issue ?

File: .github/workflows/golint.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v3
+
+            - name: Set up Go
+              uses: actions/setup-go@v3
+              with:
+                    go-version: '1.21'  
+
+            - name: Install golangci-lint
```
Comment (dr-housemd) [inline .github/workflows/golint.yaml:24]: We want all packages to be fetched and setup with bsf and Nix. 
Please check how we've used Taskfile and GitHub Actions to perform a build/test. 
So , 
* Add golangci-lint to dev packages (HCL block). 
* Update taskfile to run it
* Update GitHub Actions to invoke the taskfile .

Comment (dr-housemd) [reply .github/workflows/golint.yaml:24]: PS: we do need a dedicated workflow file, we can re-use the existing workflow file

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -14,14 +14,33 @@ jobs:
...
-        run : task build
-        continue-on-error: true
\ No newline at end of file
+
+      - name: Build app
+        run: task build
+
+      - name: Run GolangCI-Lint
+        run: task lint
+        continue-on-error: true
```
Comment (dr-housemd) [inline .github/workflows/test.yml:46]: I'd actually remove this. We want to fail if the linter fails.

Comment (dr-housemd) [reply .github/workflows/test.yml:46]: Another PR wil be created for this as per discussion on discord

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -14,14 +14,33 @@ jobs:
...
+      - name: Setup Nix cache
+        uses: DeterminateSystems/magic-nix-cache-action@main
+
+      - name: Setup Nix development environment
+        uses: nicknovitski/nix-develop@v1
         with:
           arguments: bsf/.#devShell
-      - name: run tests
+
+      - name: Set up Go
```
Comment (dr-housemd) [inline .github/workflows/test.yml:29]: We don't need this actually. We have all these dependencies defined in bsf.hcl, so its possible to access these binaries directly without adding new actions.

Comment (hanshal101) [reply .github/workflows/test.yml:29]: removing the entire Go Setup?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -14,14 +14,28 @@ jobs:
...
+      - name: Setup Nix cache
+        uses: DeterminateSystems/magic-nix-cache-action@main
+
+      - name: Setup Nix development environment
+        uses: nicknovitski/nix-develop@v1
         with:
           arguments: bsf/.#devShell
-      - name: run tests
+
+      - name: Install Task
```
Comment (dr-housemd) [inline .github/workflows/test.yml:29]: this is not needed either. Same reason. This is available from bsf develop shell

Comment (hanshal101) [reply .github/workflows/test.yml:29]: Sorry didn't saw this `go-task@~3.37.2`

**Review by dr-housemd — APPROVED** [review]

**Pull request merged**